.github/workflows/valid-files.yml

```diff
@@ -0,0 +1,55 @@
+name: Check if files are valid
+
+on:
+  push:
+  pull_request:
+
+jobs:
+  max-file-size:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Fetch everything
+        run: git fetch --prune --unshallow
+
+      - name: Error if any file is above 1 MB
+        run: |
+          FILES="$(git diff --name-only origin/master...)"
+
+          for FILE in $FILES; do
+            if [ "$(stat --printf="%s" "$FILE")" -gt 100000 ]; then
+              echo "$FILE is larger than 10 KB"
+              echo "Please ping @iHiD if you are sure that this file is required."
+              exit 1
+            fi
+          done
+
+  valid-filenames:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Fetch everything
+        run: git fetch --prune --unshallow
+
+      - name: Error if filenames contain illegal characters or words
+        run: |
+          FILES="$(git diff --name-only origin/master...)"
```

Comment:
This will also return deleted files, so would that mean that one wouldn't be able to fix a wrongly named file? :)

Reply:
I didn't consider that case. I thought that we'd check the entire repo for wrong files and then don't allow new files that don't follow the rules. If one adds and then deletes a file, it won't be returned by this command, will it?